### DISCUSSION TITLE
Custom hostname and white labelling

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#1671] [Ideas]
**Why**
Since we are open source we should allow people to make the CRM their own, even if they don't self host.
They should be able to access it under a custom domain name, or even provide a customized version to their users.
E.g. someone that wants to create a CRM for real estate

**Basic implementation**
- Create entity Domains (ex: slug, cname, logo, owning_workspace_id)
- Create Many-to-Many relationship between Domains and Workspace
- Adapt signup/login: Display logo and name
- Adapt signup: when someone signs up from a url, automatically attach to domain
- Adapt login: when someone logins, only look for workspaces/users with matching domains

**Long-term**
Once we have more options to configure the way the app works at the workspace level (e.g. custom layout, custom objects, apps, config, etc.). We'll want these settings to be attachable not only to the workspace but also the Domain-level. 
A custom object attached to the domain-level becomes similar to a standard object for a non-owner workspace user.

**Comment by BernardMcWeeney** [comment]
Curious is this currently possible on self host? I cant seem to be able to do this. I have to use ipaddress:port

**Reply by FelixMalfait** [reply]
Custom hostname is possible on self-host yes! But you can't whitelabel/customize the UI.
You should look into adding a proxy server like nginx

**Reply by JarWarren** [reply]
What would it take to support white labeling? I might be able to contribute, depending on scope/blockers.

**Reply by FelixMalfait** [reply]
Honestly it's not really a priority for us, we don't have any design for this yet so it's unlikely we would merge this sorry. There's an issue backlog full of bugs and feature requests we are focused on for now 😁

**Reply by SurajRaika** [reply]
Any update 
If are moving in that direction then 
You cN charge money for white labeling or something

**Reply by themodev** [reply]
any updates on white labeling? i'll gladly pay

**Reply by FelixMalfait** [reply]
Our top priority is the extensibility framework which will allow you to build anything on top of Twenty cleanly, we're releasing that in March. Then most likely white labelling comes back on the table again, but we need to focus on extensibility + stability first

**Reply by vivek-pk** [reply]
> any updates on white labeling? i'll gladly pay
we can help you with this, connect with us at info@coddletech.com
